src/main/java/at/hannibal2/skyhanni/features/itemabilities/CrownOfAvariceCounter.kt

```diff
@@ -106,6 +106,11 @@ object CrownOfAvariceCounter {
         update()
     }
 
+    fun isAvariceConsuming(): Boolean {
+        if (!isWearingCrown) return false
+        return totalCoins?.toDouble() != MAX_AVARICE_COINS
+    }
```

Comment:
```suggestion
    fun isAvariceConsuming(): Boolean =
        isWearingCrown && (totalCoins ?: 0) < MAX_AVARICE_COINS
```
Cleaner code, plus `<` check probably better in case they ever pull some shenanigans with it being over max somehow.